src/FunctionalExtras.Tests/Data/ValidationTests.cs

```diff
@@ -0,0 +1,707 @@
+          IValidation<string, Guid> testFirst = Validation.Failure<string, Guid>(testValue1);
+          IValidation<string, Guid> expectedResult = Validation.Failure<string, Guid>(new List<string>
+          {
+            testValue1,
+            testValue2
+          });
+          IValidation<string, Guid> actualResult = Validation.Concat(testSecond)(testFirst);
+
+          Assert.Equal(expectedResult, actualResult);
+        }
+      }
+
+      public class DescribeFailures
+      {
+        [Fact]
+        public void ShouldReturnEmptyEnumerableForNullEnumerable()
+        {
+          IEnumerable<IValidation<string, Guid>> testEnumerable = null;
+          IEnumerable<string> expectedResult = Enumerable.Empty<string>();
+          IEnumerable<string> actualResult = Validation.Failures(testEnumerable);
+
+          Assert.Equal(expectedResult, actualResult);
+        }
+
+        [Fact]
+        public void ShouldReturnEmptyEnumerableForEmptyEnumerable()
+        {
+          IEnumerable<IValidation<string, Guid>> testEnumerable = Enumerable.Empty<IValidation<string, Guid>>();
+          IEnumerable<string> expectedResult = Enumerable.Empty<string>();
+          IEnumerable<string> actualResult = Validation.Failures(testEnumerable);
+
+          Assert.Equal(expectedResult, actualResult);
+        }
+
+        [Fact]
+        public void ShouldReturnEmptyEnumerableForBlankEnumerable()
+        {
+          IEnumerable<IValidation<string, Guid>> testEnumerable = new List<IValidation<string, Guid>> { null };
+          IEnumerable<string> expectedResult = Enumerable.Empty<string>();
+          IEnumerable<string> actualResult = Validation.Failures(testEnumerable);
+
+          Assert.Equal(expectedResult, actualResult);
+        }
+
+        [Fact]
+        public void ShouldReturnEnumerableOfFalureValuesForMixedEnumerable()
+        {
+          string testFailureValue1 = Guid.NewGuid().ToString();
+          string testFailureValue2 = Guid.NewGuid().ToString();
+          Guid testSuccessValue1 = Guid.NewGuid();
+          Guid testSuccessValue2 = Guid.NewGuid();
+          IEnumerable<IValidation<string, Guid>> testEnumerable = new List<IValidation<string, Guid>>
+          {
+            Validation.Failure<string, Guid>(testFailureValue1),
+            Validation.Failure<string, Guid>(testFailureValue2),
+            Validation.Success<string, Guid>(testSuccessValue1),
+            Validation.Success<string, Guid>(testSuccessValue2)
+          };
+          IEnumerable<string> expectedResult = new List<string>
+          {
+            testFailureValue1,
+            testFailureValue2
+          };
+          IEnumerable<string> actualResult = Validation.Failures(testEnumerable);
+
+          Assert.Equal(expectedResult, actualResult);
+        }
+      }
+
+      public class DescribeFromFailure
+      {
+        private static IEnumerable<string> _testDefaultValues = new List<string>
```

Comment:
Consider making `readonly`.

Reply:
Done and done. I did it in the other tests. Not sure why I missed it here.